File: src/main/java/tn/esprit/devdream/services/FeedbackService.java
package tn.esprit.devdream.services;

import tn.esprit.devdream.entities.Feedback;

import java.util.List;
import java.util.Map;

public interface FeedbackService {
    List<Feedback> retrieveAllfeedbacks();

    Feedback retrieveFeedback(Long id_feedback);

    Feedback addFeedback(Feedback feedback);

    Map<String, Long> getStatisticsfeedback();

    Feedback updateFeedbackRating(Long idFeedback, int rating);
}

File: src/app/ranim/feedback/feedback.service.ts
import { Injectable } from '@angular/core';
import { Feedback } from './feedback/feedback.module';
import { Observable } from 'rxjs';
import { HttpClient } from '@angular/common/http';

@Injectable({
  providedIn: 'root'
})
export class FeedbackService {
  private apiURL = 'http://localhost:9000/DevDream/feedback';
  constructor(private http: HttpClient) { }
  addFeedback(feedbackData: Feedback): Observable<any>{
    return this.http.post<any>(this.apiURL+"/add-feedback", feedbackData);
  }
  getFedbacks(): Observable<any> {
    return this.http.get<Feedback[]>(`${this.apiURL}/retrieve-all-feedback`);
  }

  retrieveFeedback(id_feedback: number): Observable<any> {
    return this.http.get<any>(`${this.apiURL}/retrieve-feedback/${id_feedback}`);
  }
  updateFeedbackRating(id_feedback: number, rating: number): Observable<any> {
    return this.http.put<any>(`${this.apiURL}/update-feedback-rating/${id_feedback}`, { rating });
  }
}
